src/structs/Layer.jl

```diff
@@ -0,0 +1,140 @@
+"""
+    Layer
+
+Defines a new layer withing the video.
+
+# Fields
+- `frames::Frames`: A range of frames for which the `Layer` exists
+- `width::Int`: Width of the layer
+- `height::Int`: hegiht of the layer
+- `position::Point`: initial positon of the center of the layer on the main canvas
+- `children::Vector{AbstractObject}`: Objects defined under the layer
+- `actions::Vector{AbstractAction}`: a list of actions applied to the entire layer 
+- `current_setting::LayerSetting`: The current state of the layer see [`LayerSetting`](@ref)
+- `opts::Dict{Symbol,Any}`: can hold any options defined by the user
+- `image_matrix::Vector`: Hold the Drwaing of the layer as a Luxor image matrix
+"""
+mutable struct Layer <: AbstractObject
+    frames::Frames
+    width::Int
+    height::Int
+    position::Point
+    children::Vector{AbstractObject}
+    actions::Vector{AbstractAction}
+    current_setting::LayerSetting
+    opts::Dict{Symbol,Any}
+    image_matrix::Vector
+end
+
+"""
+    CURRENT_LAYER
+
+holds the current layer in an array to be declared as a constant
+The current layer can be accessed using CURRENT_LAYER[1]
+"""
+const CURRENT_LAYER = Array{Layer,1}()
+
+"""
+    PUSH_TO_LAYER
+
+A setinel to aid the creation of layers.
+If set to true, all the objects are pushed to the current layer.
+Can be accessed using PUSH_TO_LAYER[1]
+"""
+const PUSH_TO_LAYER = Array{Any,1}()
+# by default push all the objects to the video
+push!(PUSH_TO_LAYER, false)
+
+# for width, height and position defaults are defined in the to_layer_m function
+function Layer(
+    frames,
+    width,
+    height,
+    position;
+    children = AbstractObject[],
+    actions = AbstractAction[],
+    setting = LayerSetting(),
+    misc = Dict{Symbol,Any}(),
+    mat = Any[nothing],
+)
+    layer = Layer(frames, width, height, position, children, actions, setting, misc, mat)
+
+    if isempty(CURRENT_LAYER)
+        push!(CURRENT_LAYER, layer)
+    else
+        CURRENT_LAYER[1] = layer
+    end
+    push!(CURRENT_VIDEO[1].layers, layer)
+
+    return layer
+end
+
+"""
+    Javis.@Layer(frames, width, height, position, body)
```

Comment:
I think that this should be named something like `@JLayer` per @Wikunia . I am wondering if we are using the wrong mental model of this functionality. Because a Layer doesn't quite capture I think what all this functionality does. But I can't think of a better word for this as it describes a group of objects in a layer. So maybe we could call it "Group", "Grouping", "Group Layer", or something along those lines? Maybe `@grouped_layer` such that we could semantically speak about it with our current thinking of Action-Object syntax and say that this "creates a group of objects in a layer".

Reply:
Let's go with `JLayer` for now and let the layers idea mature more?

Reply:
I think layer makes a lot of sense besides the problem that Luxor is already using it. I'm fine with `JLayer`